src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -142,28 +142,30 @@ public string VerifyRawTransaction(string rawTx)
         }
 
         [JsonRpcMethod("eth_getTransactionByBlockHashAndIndex")]
-        public JObject? GetTransactionByBlockHashAndIndex(string blockHash, ulong index)
+        public JObject? GetTransactionByBlockHashAndIndex(string blockHash, string index)
         {
+            ulong txIndex = index.HexToUlong();
             var block = _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHash(blockHash.HexToUInt256());
             if (block is null)
                 return null;
-            if ((int)index >= block.TransactionHashes.Count)
+            if ((int)txIndex >= block.TransactionHashes.Count)
                 return null;
-            var txHash = block.TransactionHashes[(int)index];
+            var txHash = block.TransactionHashes[(int)txIndex];
             var receipt = _stateManager.LastApprovedSnapshot.Transactions.GetTransactionByHash(txHash);
             return receipt is null ? null : Web3DataFormatUtils.Web3Transaction(receipt!, block?.Hash, receipt.Block);
         }
 
         [JsonRpcMethod("eth_getTransactionByBlockNumberAndIndex")]
-        public JObject? GetTransactionByBlockNumberAndIndex(string blockTag, ulong index)
+        public JObject? GetTransactionByBlockNumberAndIndex(string blockTag, string index)
         {
+            ulong txIndex = index.HexToUlong();
```

Comment:
The same,  are you sure index is always hex string?

Reply:
https://infura.io/docs/ethereum/json-rpc/eth-getTransactionCount here is the details, where it is accepting the hex string as a input